src/test/java/org/apache/commons/cli/OptionsTest.java

```diff
@@ -78,6 +79,26 @@ public void testGetOptionsGroups() {
         assertEquals(2, options.getOptionGroups().size());
     }
 
+    @Test
+    public void testAddOptions() {
+        final Options options = new Options();
+
+        final OptionGroup group1 = new OptionGroup();
+        group1.addOption(Option.builder("a").build());
+        group1.addOption(Option.builder("b").build());
+
+        options.addOptionGroup(group1);
+
+        options.addOption(Option.builder("X").build());
+        options.addOption(Option.builder("y").build());
+
+        final Options underTest = new Options();
+        underTest.addOptions(options);
```

Comment:
You need a test that calls `addOptions()` twice with different options to verify that the calls are cumulative and do not replace what's already there. 

Also: What should happen if I add the same options twice? For example:
```
underTest.addOptions(options);
underTest.addOptions(options);
```
Also test adding an empty Options object to make sure nothing odd happens. 

Edge cases, edge cases... ;-)